chapter3/src/order.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+type Order struct {
+	Id          string
+	orderLines  []OrderLine
+	total       float32
+	dateCreated time.Time
+	user        string
+}
+
+type OrderLine struct {
+	Id          string
+	item        string
+	dateCreated time.Time
+	quantity    int16
+	unitPrice   float32
+}
```

Comment:
Typically you don't define types in main package. This package is only meant for the code that runs your program. You could move these types into a dedicated package, which becomes more important as your codebase grows.

Reply:
Assuming you do introduce a new package for orders, the `CreateOrder` method would become a public `New` method in that package.

Reply:
Ohh thats true, thnx!